examples/04_manipulating_images/plot_roi_extraction.py

```diff
@@ -1,131 +1,282 @@
-plot_stat_map(new_img_like(fmri_img, log_p_values),
-              mean_img, title="p-values", cut_coords=cut_coords)
+
+# Before visualizing, we transform the computed p-values to Nifti-like image
+# using function `new_img_like` from nilearn. Otherwise, an error will be
+# raised.
+from nilearn.image import new_img_like
+
+# First argument being a reference image and second argument should be p-values
+# data to convert to a new image as output. This new image will have same header
+# information as reference image.
+log_p_values_img = new_img_like(fmri_img, log_p_values)
+
+# Now, we visualize log p-values image on functional mean image as background
+# with coordinates given manually and colorbar on the right side of plot (by
+# default colorbar=True)
+plot_stat_map(log_p_values_img, mean_img,
+              title="p-values", cut_coords=cut_coords)
 
 ##############################################################################
-# Build a mask from this statistical map
+# Build a mask from this statistical map (Improving the quality of the mask)
+# --------------------------------------------------------------------------
+# **Thresholding** - Voxels with lower p-values are kept as voxels of interest.
+# Applying a threshold to an array is easy thanks to numpy indexing à la Matlab.
 
-# Thresholding
+# Note that we use log p-values data; we force values below 5 to 0 by
+# thresholding.
 log_p_values[log_p_values < 5] = 0
-plot_stat_map(new_img_like(fmri_img, log_p_values),
-              mean_img, title='Thresholded p-values', annotate=False,
+
+# Visualize the reduced voxels of interest using statistical image plotting
+# function. As shown above, we first transform data in array to Nifti image.
+log_p_values_img = new_img_like(fmri_img, log_p_values)
+
+# Now, visualizing the created log p-values to image without colorbar and
+# without Left - 'L', Right - 'R' annotation
+plot_stat_map(log_p_values_img, mean_img,
+              title='Thresholded p-values', annotate=False,
               colorbar=False, cut_coords=cut_coords)
 
 ##############################################################################
-# Binarization and intersection with VT mask
-# (intersection corresponds to an "AND conjunction")
+# **Binarization** and **Intersection** with Ventral Temporal (VT) mask - We
+# now want to restrict our investigation to the VT area. The corresponding
+# spatial mask is provided in haxby_dataset.mask_vt. We want to compute the
+# intersection of this provided mask with our self-computed mask.
+
+# self-computed mask
 bin_p_values = (log_p_values != 0)
+# VT mask
 mask_vt_filename = haxby_dataset.mask_vt[0]
-import nibabel
-vt = nibabel.load(mask_vt_filename).get_data().astype(bool)
+
+# The first step is to load VT mask with nilearn's utilities check_niimg
+# function and same time convert data type numbers to boolean type
+from nilearn._utils import check_niimg
+
+vt = check_niimg(mask_vt_filename).get_data().astype(bool)
```

Comment:
> Actually, I think that it would be a good thing to make masking._load_mask_img
> public. On top of loading mask from strings and converting to boolean, it make
> necessary checks like dimensionality and number of unique values.

Right, but it's a fairly specific function. And I worry about having a
too big public API: it makes things harder to find.